osquery/requests/serializers/json.h

```diff
@@ -0,0 +1,56 @@
+/*
+ *  Copyright (c) 2014, Facebook, Inc.
+ *  All rights reserved.
+ *
+ *  This source code is licensed under the BSD-style license found in the
+ *  LICENSE file in the root directory of this source tree. An additional grant
+ *  of patent rights can be found in the PATENTS file in the same directory.
+ *
+ */
+
+#pragma once
+
+#include "osquery/requests/requests.h"
+
+namespace osquery {
+
+/**
+ * @brief JSON Serializer
+ */
+class JSONSerializer : public Serializer {
```

Comment:
I like how right now, if you want to make a request, you include two files:

```cpp
#include "osquery/requests/transports/http.h" // transport 
#include "osquery/requests/serializers/json.h" // serializer
```

It's really clear from your includes what you want, what you plan on doing etc. It's also really easy for someone to see the serializers and transports, figure out to do, and add a new one. I could be convinced to move them somewhere that makes sense given their functionality, but throwing them all in `core/text.cpp` seems like hiding them.

Since `core/text.cpp` only has three functions in it right now, I think it would be fair/safe to say that whatever we decide to do here, `core/text.cpp` should follow, but I don't think `core/text.cpp` is the destination.